### PR TITLE
aws profiles and upload throttling

### PR DESCRIPTION
**Pull request by mattosborn**
- update aws-sdk
- add aws profile option
- use `lodash.throttle` for upload throttling
- refactor to allow multiple instances to function correctly
- added name option
- refactored into a single file

### CODE REVIEW

**Comment by lazywithclass** [comment]
Thank you! I will review this and see why it's failing on the build.
There isn't any serious testing in place, but I'd like to keep it green ;D

File: index.js [hunk 1]
```diff
@@ -1,23 +1,84 @@
...
+  this.name = options.name || 'CloudWatch';
   this.level = options.level || 'info';
+  this.logEvents = [];
+
+  this.options = options;
+
+  if (options.awsProfile && options.awsRegion) {
+    AWS.config.credentials = new AWS.SharedIniFileCredentials({profile: options.awsProfile});
+    this.cloudwatchlogs = new AWS.CloudWatchLogs({region: options.awsRegion});
+  } else if (awsAccessKeyId && awsSecretKey && options.awsRegion) {
```
Comment (lazywithclass) [inline index.js]: Is this meant to be something like

    } else if (options.awsAccessKeyId && options.awsSecretKey && options.awsRegion) {

or am I missing something?

File: index.js [hunk 2]
```diff
@@ -1,23 +1,84 @@
...
+    if (!logStream) throw Exception("Log stream '" + this.options.logStreamName +"' not found");
+
+    cb(err, logStream.uploadSequenceToken);
+
+  }.bind(this));
+}
+
+CloudWatch.prototype.upload = function() {
+
+  if (this.logEvents.length <= 0) { return; }
```
Comment (lazywithclass) [inline index.js]: For consistency I would remove `{` and `}` for these one line `if`s.

File: index.js (hunk 1 above)
Comment (mattosborn) [inline index.js]: Yep sorry about that. Total lack of testing on my part as I threw this together fairly quickly yesterday. Will give it a once over this morning.

Comment (lazywithclass) [inline index.js]: Ok no problems with that, once this is done I will merge, fix the tests and then release a new version on npm.

**Comment by lazywithclass** [comment]
I've manually merged this, I am testing it to see that everything is fine, I've committed as you, with me as committer.

You can check the developments here: https://github.com/lazywithclass/winston-cloudwatch/tree/improvements-to-upload

**Pull request closed without merging**